### PR TITLE
fix: support binding ipv6 address

### PR DESCRIPTION
**Pull request by thxCode**
**Problem:**

Unable to start the server with --bind-address="ipv6 address", as the string in the form of "ipv6:port" is invalid for net listening.

**Solution:**

Parse the given bind address, if the given address is a strict IPv6, we use "[ipv6]:port" format.

Allow user to pick dual-stack mode or single-stack mode now.

**Related Issue:**
#1202

### CODE REVIEW

**Review by rainfd — COMMENTED** [review]
LGTM

**Review by rainfd — DISMISSED** [review]
LGTM

File: pkg/apis/server.go [hunk 1]
```diff
@@ -222,15 +235,39 @@ func serve(ctx context.Context, handler http.Handler, errorLog *stdlog.Logger, l
 	return nil
 }
 
-func newTcpListener(ctx context.Context, ip string, port int) (net.Listener, error) {
-	address := fmt.Sprintf("%s:%d", ip, port)
+func parseBindAddress(ip string, dual bool, port int) (network, address string, err error) {
```
Comment (gitlawr) [inline pkg/apis/server.go:238]: nit: The order `ip, port, dual` looks more comfortable to me.

Can we add some unit tests?

Comment (thxCode) [reply pkg/apis/server.go]: updated

**Review by gitlawr — APPROVED** [review]
LGTM

**Pull request merged**